packages/nimble-components/src/banner/tests/banner-matrix.stories.ts

```diff
@@ -0,0 +1,100 @@
+import type { Story, Meta } from '@storybook/html';
+import { withXD } from 'storybook-addon-xd-designs';
+import { html, ViewTemplate, when } from '@microsoft/fast-element';
+import {
+    createMatrixThemeStory,
+    createStory
+} from '../../utilities/tests/storybook';
+import {
+    createMatrix,
+    sharedMatrixParameters
+} from '../../utilities/tests/matrix';
+import { hiddenWrapper } from '../../utilities/tests/hidden';
+import '../../all-components';
+import { loremIpsum } from '../../utilities/tests/lorem-ipsum';
+
+const metadata: Meta = {
+    title: 'Tests/Banner',
+    decorators: [withXD],
+    parameters: {
+        ...sharedMatrixParameters(),
+        design: {
+            artboardUrl:
+                'https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/29c405f7-08ea-48b6-973f-546970b9dbab/specs/'
+        }
+    }
+};
+
+export default metadata;
+
+const severityStates = [
+    ['Neutral', undefined],
+    ['Error', 'error'],
+    ['Warning', 'warning'],
+    ['Info', 'info']
+] as const;
+type SeverityState = typeof severityStates[number];
+
+const actionStates = [
+    ['', false, undefined],
+    ['link', true, undefined],
+    ['ghost button', false, 'ghost'],
+    ['outline button', false, 'outline']
+] as const;
+type ActionState = typeof actionStates[number];
+
+const partsHiddenStates = [
+    ['', false],
+    ['parts hidden', true]
+] as const;
+type PartsHiddenState = typeof partsHiddenStates[number];
+
+// prettier-ignore
+const component = (
+    [severityLabel, severity]: SeverityState,
+    [actionLabel, linkVisible, buttonAppearance]: ActionState,
+    [partsHiddenLabel, partsHidden]: PartsHiddenState
+): ViewTemplate => html`
+    <nimble-banner
+        open
+        severity="${severity!}"
+        title-hidden="${partsHidden}"
+        prevent-dismiss="${partsHidden}"
+    >
+        <span slot="title">${severityLabel} ${actionLabel} ${partsHiddenLabel}</span>
+        This is message text.
```

Comment:
Can we create an additional dimension in the matrix for long vs short text?

Reply:
I have a separate story for testing wrapping/long text. Do you think we need to test that with all the combinations of other settings? That would make this story a lot bigger.

Reply:
Ah, I see that now. I do think that having that be part of the matrix is more consistent with how our other matrix tests are set up, so I would lean in that direction unless it causes some other problem.

Reply:
It still feels like overkill to me (i.e. testing it in every combination of other variables), but I don't feel that strongly about it.

Reply:
I'm not sure why I just got an email notification about this comment when github says you posted it 6 hours ago, but whatever you think gives us adequate test coverage is fine with me.